src/validators/ip-validator.js

```diff
@@ -0,0 +1,54 @@
+// Implements ipv4 and ipv6 format validations as perhttps://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-7.3.4
+var ipValidator = {

```

Comment:
Shouldn't the validation code be included directly in` src/validator.js` rather than an external file?

Reply:
Hi @pmk65, thanks for the feedback. Obviously it's good practice to break code down into small functional units since it makes the code more readable and testable. When and how you choose to distribute this is a matter of preferred coding style. I took my lead from the layout of the editor code, which has a single file then a subdirectory for specific implementations. I thought this approach would make it easier for future contributors to add extra validators where necessary, but it could equally be moved back into the main file.

There's one other thing to consider though, the mechanics of unit testing: Currently the code is built by simply concatenating files according to the order specified in the gruntfile. This is obviously a nice simple approach but does make it harder to manage dependencies and also to import files for testing: When unit testing you usually `require()` or `import` the files and hence the objects you want to test but you can't do this easily with the current setup since the 'Validator` class has external dependencies not declared in `validator.js`. So I can't simply export then import the `Validator` class - I have to export a standalone function using the  `if( typeof(exports)...` construct you see in `ip-validator.js`. And that seemed to go better in a different file.

tldr: It's a pain trying to unit-test code if it isn't built using a module-loader like requirejs or using `import` and `export` statements provided in later javascript versions.

Reply:
@sirockin Placing the validators in separate files is not a bad idea. The same should also be done for the resolver codes in `src/default.js`. That way you can easily build a custom version with only the editors you need/want. But currently the editors needs an overhaul first #401 

As for the unit tests, are you using the **src/** files for the tests? Would it help using the **dist/** files? (which is a single file build from the files in` src/`) 

Is it possible to create a module-loader using the existing code base?

Reply:
Hi @pmk65 . Using the dist build may work but I wouldn't advise it since it's better to automatically run unit tests on the source code before building. (for this repo we could do it from the gruntfile) 

I'm sure the existing code base could be fairly easily adapted to use require(). You may instead want to consider upgrading the javascript version to one which allows use of import/export as I think this is the preferred way of doing things these days. Again, this shouldn't be too difficult as all the javascript versions are backwards-compatible.

Reply:
Actually `require()` is probably the best way to go (using [browserify](http://browserify.org/)) since javascript ES6 (and hence import) isn't yet supported in Internet Explorer.